doozerlib/assembly_inspector.py

```diff
@@ -0,0 +1,260 @@
+                content_git_url = rpm_meta.config.content.source.git.url
+                if content_git_url:
+                    # Make sure things are in https form so we can compare
+                    # content_git_url = util.convert_remote_git_to_https(content_git_url)
+
+                    # TODO: Download spec file associated with the build and verify SOURCE_GIT_URL
+
+                    try:
+                        target_branch = rpm_meta.config.content.source.git.branch.target
+                        if target_branch:
+                            _ = int(target_branch, 16)  # parse the name as a git commit
+                            # if we reach here, a git commit hash was declared as the
+                            # upstream source of the rpm package's content. We should verify
+                            # it perfectly matches what we find in the assembly build.
+                            # Each package build gets git commits encoded into the
+                            # release field of the NVR. So the NVR should contain
+                            # the desired commit.
+                            build_nvr = brew_build_dict['nvr']
+                            if target_branch[:7] not in build_nvr:
+                                issues.append(f'{dgk} build for rhel-{el_ver} did not find git commit {target_branch[:7]} in package RPM NVR {build_nvr}')
+                    except ValueError:
+                        # The meta's target branch a normal branch name
+                        # and not a git commit. When this is the case,
+                        # we don't try to assert anything about the build's
+                        # git commit.
+                        pass
+
+        return issues
+
+    def check_group_image_consistency(self, build_inspector: BrewBuildImageInspector) -> List[str]:
+        """
+        Evaluate the current assembly build and an image in the group and check whether they are consistent with
+        :param build_inspector: The brew build to check
+        :return: Returns a (potentially empty) list of reasons the image should be rebuilt.
+        """
+        image_meta = build_inspector.get_image_meta()
+        self.runtime.logger.info(f'Checking group image for consistency: {image_meta.distgit_key}...')
+        issues: List[str] = []
+
+        installed_packages = build_inspector.get_all_installed_package_build_dicts()
+
+        """
+        If the assembly defined any RPM package dependencies at the group or image
+        member level, we want to check to make sure that installed RPMs in the
+        build image match the override package.
+        If reading this, keep in mind that a single package/build may create several
+        RPMs. Both assemblies and this method deal with the package level - not
+        individual RPMs.
+        """
+        package_overrides: Dict[str, str] = image_meta.get_assembly_rpm_package_dependencies(el_ver=image_meta.branch_el_target())
```

Comment:
Why `build_inspector.get_rhel_base_version()` is not used here to determine RHEL version?

Reply:
`get_rhel_base_version` is generic and meant to work even when we don't have `image_meta` for the image. The `image_meta` version can't return None, so I preferred it.